gnovm/pkg/gnolang/nodes.go

```diff
@@ -1651,6 +1697,21 @@ func (sb *StaticBlock) GetPathForName(store Store, n Name) ValuePath {
 	panic(fmt.Sprintf("name %s not declared", n))
 }
 
+// Get the containing block node for node with path relative to this containing block.
+func (sb *StaticBlock) GetBlockNodeForPath(store Store, path ValuePath) BlockNode {
```

Comment:
Can you replace the code in StaticBlock.GetStaticTypeOfAt with this method?

Reply:
done.